crypto/bls/bls.go

```diff
@@ -252,3 +252,15 @@ func SerializedSignatureFromBytes(serializedSignature []byte) (SerializedSignatu
 	copy(signatureBytesFixed[:], serializedSignature)
 	return signatureBytesFixed, nil
 }
+
+func UncompressKey(serialized SerializedPublicKey) []byte {
+	publicKey, err := bls.DeserializePublicKeyCached(serialized[:])
+	if err != nil {
+		return nil
+	}
+	uncompressedBytes, err := publicKey.SerializeUncompressed()
+	if err != nil {
+		return nil
+	}
+	return uncompressedBytes
```

Comment:
maybe here we can check that len is 192? that means we know for a fact that if we return something here it's correct.

Also, in that sense, it might make sense for this function to return an error... since `nil` doesn't really work as a good error value. I don't see any place checking if the returned value by this function is nil and reacting to that...

so we are effectively silently ignoring errors in this pr

Reply:
I'll just pass the error instead. The only place this value is currently used is in the precompile, where it checks that the length is correct (note that `len(nil) == 0`).